### PR TITLE
Cypress. Test "Rotated bounding boxes"

### PR DESCRIPTION
**Pull request by dvkruchinin**
Related to #3832 

### Motivation and context

Adding a test to test functionality "Rotated bounding boxes"

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2021 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: tests/cypress/integration/actions_objects2/case_108_rotated_bounding_boxes.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+            .should('be.visible')
+            .and('have.length', 1)
+            .trigger('mousemove')
+            .trigger('mouseover')
+            .trigger('mousedown', { button: 0 });
+        if (shift) {
+            cy.get('.cvat-canvas-container').trigger('mousemove', x, y, { shiftKey: true });
+        } else {
+            cy.get('.cvat-canvas-container').trigger('mousemove', x, y);
+        }
```
Comment (bsekachev) [inline tests/cypress/integration/actions_objects2/case_108_rotated_bounding_boxes.js:47]: ``cy.get('.cvat-canvas-container').trigger('mousemove', x, y, { shiftKey: shift });``?

File: tests/cypress/integration/actions_objects2/case_108_rotated_bounding_boxes.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        labelName,
+        firstX: createRectangleShape2Points.firstX,
+        firstY: createRectangleShape2Points.firstY - 150,
+        secondX: createRectangleShape2Points.secondX,
+        secondY: createRectangleShape2Points.secondY - 150,
+    };
+
+    const transformMatrixShape = [];
+
+    function testShapeRotate(shape, x, y, expectedRorate, shift) {
```
Comment (bsekachev) [inline tests/cypress/integration/actions_objects2/case_108_rotated_bounding_boxes.js:32]:
```suggestion
    function testShapeRotate(shape, x, y, expectedRotate, shift) {
```

File: tests/cypress/integration/actions_objects2/case_108_rotated_bounding_boxes.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+            .and('have.length', 1)
+            .trigger('mousemove')
+            .trigger('mouseover')
+            .trigger('mousedown', { button: 0 });
+        if (shift) {
+            cy.get('.cvat-canvas-container').trigger('mousemove', x, y, { shiftKey: true });
+        } else {
+            cy.get('.cvat-canvas-container').trigger('mousemove', x, y);
+        }
+        cy.get('#cvat_canvas_text_content').should('contain.text', expectedRorate);
```
Comment (bsekachev) [inline tests/cypress/integration/actions_objects2/case_108_rotated_bounding_boxes.js:48]:
```suggestion
        cy.get('#cvat_canvas_text_content').should('contain.text', expectedRotate);
```

File: tests/cypress/integration/actions_objects2/case_108_rotated_bounding_boxes.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+    describe(`Testing case "${caseId}"`, () => {
+        it('Check that bounding boxes can be rotated.', () => {
+            testShapeRotate('#cvat_canvas_shape_1', 350, 150, '11.4°');
+            testShapeRotate('#cvat_canvas_shape_2', 350, 150, '26.6°');
+            cy.get('#cvat_canvas_shape_2').then((shape2) => {
+                const shapeAttrs = shape2.attr('transform').split('(')[1].split(')')[0].split(',');
+                for (const i of shapeAttrs) {
+                    transformMatrixShape.push(Number(i).toFixed(3));
+                }
```
Comment (bsekachev) [inline tests/cypress/integration/actions_objects2/case_108_rotated_bounding_boxes.js:69]: Try to use getCTM() instead.

File: tests/cypress/integration/actions_objects2/case_108_rotated_bounding_boxes.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+            // A single click does not reproduce the split a track scenario in cypress test.
+            cy.get('#cvat_canvas_shape_2').click().click();
+            cy.get('#cvat_canvas_shape_3').should('have.attr', 'transform').then((shapeTransform) => {
+                cy.get('#cvat_canvas_shape_4').should('have.attr', 'transform', shapeTransform);
+            });
+        });
+
+        it('Check rotation with hold Shift button.', () => {
+            cy.goCheckFrameNumber(0);
+            testShapeRotate('#cvat_canvas_shape_1', 350, 150, '13.0°', true);
```
Comment (bsekachev) [inline tests/cypress/integration/actions_objects2/case_108_rotated_bounding_boxes.js:125]: With shift it is not expected you can rotate to 13 degrees, the nearest is 15 degrees.

File: tests/cypress/integration/actions_objects2/case_108_rotated_bounding_boxes.js [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+            testShapeRotate('#cvat_canvas_shape_1', 350, 180, '14.2°', true);
```
Comment (bsekachev) [inline tests/cypress/integration/actions_objects2/case_108_rotated_bounding_boxes.js:126]: The same comment

File: tests/cypress/integration/actions_objects2/case_108_rotated_bounding_boxes.js [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+                });
+            }
+            cy.goCheckFrameNumber(2);
+
+            // Split tracks
+            cy.get('.cvat-split-track-control').click();
+            // A single click does not reproduce the split a track scenario in cypress test.
+            cy.get('#cvat_canvas_shape_2').click().click();
+            cy.get('#cvat_canvas_shape_3').should('have.attr', 'transform').then((shapeTransform) => {
+                cy.get('#cvat_canvas_shape_4').should('have.attr', 'transform', shapeTransform);
```
Comment (bsekachev) [inline tests/cypress/integration/actions_objects2/case_108_rotated_bounding_boxes.js:119]: One of shapes (3 or 4) should be invisible. Is it correct to try comparing their attributes?

File: tests/cypress/integration/actions_objects2/case_108_rotated_bounding_boxes.js [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+                    const transformMatrix = [];
+                    for (const i of transform) {
+                        transformMatrix.push(Number(i).toFixed(3));
+                    }
+                    expect(transformMatrix).to.deep.eq(transformMatrixShape);
+                });
+            }
+            // Merge tracks
+            cy.get('.cvat-merge-control').click();
+            cy.get('#cvat_canvas_shape_2').click();
```
Comment (bsekachev) [inline tests/cypress/integration/actions_objects2/case_108_rotated_bounding_boxes.js:88]: You didn't merge anything here, just select one existing track, right?

File: tests/cypress/integration/actions_objects2/case_108_rotated_bounding_boxes.js (hunk 2 above)
Comment (dvkruchinin) [reply tests/cypress/integration/actions_objects2/case_108_rotated_bounding_boxes.js]: Fixed.

File: tests/cypress/integration/actions_objects2/case_108_rotated_bounding_boxes.js (hunk 3 above)
Comment (dvkruchinin) [reply tests/cypress/integration/actions_objects2/case_108_rotated_bounding_boxes.js]: Fixed.

File: tests/cypress/integration/actions_objects2/case_108_rotated_bounding_boxes.js (hunk 5 above)
Comment (dvkruchinin) [reply tests/cypress/integration/actions_objects2/case_108_rotated_bounding_boxes.js]: Fixed

File: tests/cypress/integration/actions_objects2/case_108_rotated_bounding_boxes.js (hunk 6 above)
Comment (dvkruchinin) [reply tests/cypress/integration/actions_objects2/case_108_rotated_bounding_boxes.js]: Fixed

File: tests/cypress/integration/actions_objects2/case_108_rotated_bounding_boxes.js (hunk 7 above)
Comment (dvkruchinin) [reply tests/cypress/integration/actions_objects2/case_108_rotated_bounding_boxes.js]: This is probably normal. It is not necessary to make the shape visible. Or is another check expected here?

File: tests/cypress/integration/actions_objects2/case_108_rotated_bounding_boxes.js (hunk 8 above)
Comment (dvkruchinin) [reply tests/cypress/integration/actions_objects2/case_108_rotated_bounding_boxes.js]: Sure. Thanks. Fixed.
Added merge shapes and verification by analogy with tracks.

File: tests/cypress/integration/actions_objects2/case_108_rotated_bounding_boxes.js (hunk 4 above)
Comment (bsekachev) [reply tests/cypress/integration/actions_objects2/case_108_rotated_bounding_boxes.js:92]: Try to use invoke. getCTM is a DOM method, I am not sure how to call it in cypress, but maybe with ``invoke``

Comment (dvkruchinin) [reply tests/cypress/integration/actions_objects2/case_108_rotated_bounding_boxes.js]: Unfortunately, I haven't found a solution yet for ```getCTM()```.
I have attempted to convert the matrix values from attr to ```Number``` values due to a slight change in these values if the test is run in the browser in headless mode. But now I've redone it to use values from css. This solution works well.

**Review by bsekachev — APPROVED** [review]

**Pull request merged**